src/lsp_data.rs

```diff
@@ -80,6 +87,16 @@ pub mod ls_util {
         }
     }
 
+    pub fn location_from_racer_match(mtch: racer::Match) -> Option<Location> {
+        let source_path = &mtch.filepath;
+
+        if let Some(racer::Coordinate { line, column }) = mtch.coords {
+            Some(location_from_position(source_path, line - 1, column))
+        } else {
+            None
+        }
+    }
+
```

Comment:
The VFS is only dependent on rls-analysis for a definition of `Span` and there is a PR to remove that, so I don't think it is too far off what you want.